airflow/serialization/serializers/builtin.py

```diff
@@ -0,0 +1,59 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, cast
+
+from airflow.utils.module_loading import qualname
+
+if TYPE_CHECKING:
+    from airflow.serialization.serde import U
+
+__version__ = 1
+
+serializers = [frozenset, set, tuple]
+deserializers = serializers
+stringifiers = serializers
+
+
+def serialize(o: object) -> tuple[U, str, int, bool]:
+    return list(cast(list, o)), qualname(o), __version__, True
+
+
+def deserialize(classname: str, version: int, data: list) -> tuple | set | frozenset:
+    if version > __version__:
+        raise TypeError("serialized version is newer than class version")
+
+    if classname == qualname(tuple):
+        return tuple(data)
+
+    if classname == qualname(set):
+        return set(data)
+
+    if classname == qualname(frozenset):
+        return frozenset(data)
```

Comment:
```python
for t in stringifiers:
    if qualname(t) == classname:
        return t(data)
```

Does this make sense?

Reply:
The intention is not to arbitrarily instantiate a class and only allow `[stringifiers]`. I'm not entirely happy yet with the approach, so if you have a better idea I am all ears :-)